test/conftest.py

```diff
@@ -37,6 +43,36 @@ def process_struct(obj):
         raise TypeError(f"Not sure how to deal with type {type(obj)}: {obj!r}")
 
 
+def parse_example_file(dialect, sqlfile):
```

Comment:
Extracted these functions from `test/generate_parse_fixture_yml.py`, which are now used both there and by `test/dialects/dialects_test.py::test__dialect__base_parse_struct`.